src/charm.py

```diff
@@ -99,7 +99,15 @@ def __init__(self, *args):
             source_type="loki",
             source_url=self._external_url,
         )
-        scrape_jobs = [{"static_configs": [{"targets": [f"*:{self._port}"]}]}]
+
+        scrape_url = urlparse(self._external_url)
+        logger.debug("Using url {} for Prometheus scrape".format(urlunparse(scrape_url)))
```

Comment:
I [would remove](https://juju.is/docs/sdk/logging) the `.format(...`

```suggestion
        logger.debug("Using url %s for Prometheus scrape", urlunparse(scrape_url))
```

Reply:
Logger already does it, basically. It's an extremely marginal performance difference, but `logger.level("....")` has an implicit `printf`-style interpolation.

Reply:
Yes, that's true. In this case, it's kind of a micro-optimization which is just down to style. The additional ops from "double interpolating" isn't going to be the end of the world

Reply:
Closing this comment in the interest of releasing on time.